code/__HELPERS/text.dm

```diff
@@ -438,3 +438,91 @@ var/list/binary = list("0","1")
+//in the log file and have it be reflected in the in game item/mob it came from.
+//(That's what things like savefiles are for) Note that this list is not shuffled.
+/proc/twitterize(list/proposed, filename, cullshort = 0, storemax = 1000)
+	if(!islist(proposed) || !filename)
+		return
+
+	//Regular expressions are, as usual, absolute magic
+	var/regex/is_website = new("http|www.|\[a-z0-9_-]+.(com|org|net|mil|edu)+", "i")
+	var/regex/is_email = new("\[a-z0-9_-]+@\[a-z0-9_-]+.\[a-z0-9_-]+", "i")
+	var/regex/alphanumeric = new("\[a-z0-9]+", "i")
+	var/regex/punctuation = new("\[.!?]+", "i")
+
+	var/list/accepted = list()
+	for(var/string in proposed)
+		if(findtext(string,is_website) || findtext(string,is_email))
+			continue
+		var/buffer = ""
+		var/early_culling = TRUE
+		for(var/pos = 1, pos != lentext(string), pos++)
+			var/let = copytext(string, pos, (pos + 1) % lentext(string))
+			if(early_culling && !findtext(let,alphanumeric))
+				continue
+			early_culling = FALSE
+			buffer += let
+		var/punctbuffer = ""
+		var/cutoff = lentext(buffer)
+		for(var/pos = lentext(buffer), pos != 0, pos--)
+			var/let = copytext(buffer, pos, (pos + 1) % lentext(buffer))
+			if(findtext(let,alphanumeric))
+				break
+			if(findtext(let,punctuation))
+				punctbuffer = let + punctbuffer //Note this isn't the same thing as using +=
+				cutoff = pos
+		if(punctbuffer) //We clip down excessive punctuation to get the letter count lower and reduce repeats
+			var/exclaim = FALSE
+			var/question = FALSE
+			var/periods = 0
+			for(var/pos = lentext(punctbuffer), pos != 0, pos--)
+				var/punct = copytext(buffer, pos, (pos + 1) % lentext(buffer))
+				if(!exclaim && punct == "!")
+					exclaim = TRUE
+				if(!question && punct == "?")
+					question = TRUE
+				if(!exclaim && !question && punct == ".")
+					periods += 1
+			if(exclaim)
+				if(question)
+					punctbuffer = "?!"
+				else
+					punctbuffer = "!"
+			else if(question)
+				punctbuffer = "?"
+			else if(periods)
+				if(periods >= 3)
+					punctbuffer = "..."
+				else
+					punctbuffer = "" //Grammer nazis be damned
+			buffer = copytext(buffer, 1, cutoff) + punctbuffer
+
+		if(!buffer || lentext(buffer) > 140 || lentext(buffer) <= cullshort || buffer in accepted) //second to fourth gate
+			continue
+
+		accepted += buffer
+
+	var/log = file("data/npc_saves/[filename].log") //If this line ever shows up as changed in a PR be very careful you aren't being memed on
+	var/list/oldentries = file2list(log)
+	if(!isemptylist(oldentries))
+		for(var/string in accepted) //The fifth and final gate
+			for(var/old in oldentries)
+				if(string == old)
+					oldentries.Remove(old) //Line's position in line is "refreshed" until it falls off the in game radar
+					break
+
+	var/list/finalized = list()
+	for(var/list/L in list(accepted, oldentries)) //we keep old and unreferenced phrases near the bottom for culling
+		finalized += listclearnulls(L.Copy())

```

Comment:
Though accepted is definitely populated and oldentries sometimes is I've yet to find a way to transfer them to finalized and pass them through the mutator listclearnulls() that doesn't blank out the whole of finalized.

Besides this everything works.